### ISSUE TITLE
subsite ignores language

### ISSUE THREAD

**Issue by cloph** [silverstripe/silverstripe-subsites#83]
in ss 3.1, configure two subsites, one in english, one in german. your site's default lang is english.

Despite having configured the german subsite to use "German" as language, the created HTML (the simple theme uses the field $ContentLocale) is set to english (en-US)

**Comment by gregsmirnov** [comment]
I use this extension to SiteTree.
```yml
---
Name: mysubsites
Only:
  moduleexists: subsites
---
SiteTree:
  extensions:
    - SubsitesLanguage
```
```php
class SubsitesLanguage extends Extension
{
	public function contentcontrollerInit() {
		$subsite = Subsite::currentSubsite();
		if (i18n::validate_locale($subsite->Language)) {
			i18n::set_locale($subsite->Language);
		}
	}
}
```

**Comment by robbieaverill** [comment]
@gregsmirnov if this is still a problem would you be willing to submit a pull request to fix it?

**Comment by beerbohmdo** [comment]
Translation with subsite is really ugly, especially with multiple themes ...

I had to use a RequestFilter to set the theme as soon as possible to load the correct translation.
And I have to add a custom Cache Backend because the Cache from each themes overrides each other :/.

```php
<?php

class SubSiteRequestFilter implements RequestFilter
{
    public function preRequest(SS_HTTPRequest $request, Session $session, DataModel $model)
    {
        /** @var Subsite $subSite */
        if (strpos($_SERVER['REQUEST_URI'], 'dev/build') === false && ($subSite = Subsite::currentSubsite()) && $subSite->Theme) {
            Config::inst()->update('SSViewer', 'theme', $subSite->Theme);

            SS_Cache::add_backend('i18n', 'File', array(
                'file_name_prefix'  => 'zend_cache_'.$subSite->Theme,
            ));

            SS_Cache::pick_backend('i18n', 'i18n', 10);
        }
    }

    public function postRequest(SS_HTTPRequest $request, SS_HTTPResponse $response, DataModel $model)
    {
        // TODO: Implement postRequest() method.
    }
}
```

**Comment by robbieaverill** [comment]
Fixed with #288, and available in version 1.3.4
